### PR TITLE
file-server: fix %file-server by adding SPA flag and threading is-file through request

### PR DESCRIPTION
**Pull request by tacryt-socryp**
…h is-file through request

### CODE REVIEW

**Comment by matildepark** [comment]
Tested with create-landscape-app, patched for the new boolean. I'm getting 404s on the image files, but they're definitely in app/testing/img/.

image

File: pkg/arvo/app/file-server.hoon [hunk 1]
```diff
@@ -113,14 +122,17 @@
...
+        (~(put by serving) url-base clay+clay-base.act public.act spa.act)
+      ==
     ::
         %serve-glob
       =*  url-base  url-base.act
       ?:  (~(has by serving) url-base)
         ~|("url already bound to {<(~(got by serving) url-base.act)>}" !!)
       :-  [%pass url-base %arvo %e %connect [~ url-base] %file-server]~
-      this(serving (~(put by serving) url-base glob+glob.act public.act))
+      this(serving (~(put by serving) url-base glob+glob.act public.act %.y))
```
Comment (matildepark) [inline pkg/arvo/app/file-server.hoon:135]: So we pass `%.y` as the spa flag on `:glob`, which won't have the file — it mounts its state as `bundle.[hash].js`. Given the `?&` for spa and is-file flags, will this play nice?

Comment (tacryt-socryp) [reply pkg/arvo/app/file-server.hoon:135]: It works fine so long as the file name is lowercase. This is intended behavior given constraints around clay paths.

Comment (matildepark) [reply pkg/arvo/app/file-server.hoon:135]: I just mean because glob doesn't put it in clay, it pokes file-server to mount directly from state AFAIK.

Comment (tacryt-socryp) [reply pkg/arvo/app/file-server.hoon:135]: Not a concern given that is-file overrides that flag.

**Pull request merged**